### ISSUE TITLE
Breaking Ground Storage Unit Inventory Icons not Displaying

### ISSUE THREAD

**Issue by Joker58th** [shadowmage45/TexturesUnlimited#68]
Title says it all...I think.

Clean KSP 1.7.2 with both expansions
TU 1.4.8.22
-force-glcore

![TU1](https://user-images.githubusercontent.com/32574965/60289304-1bed1b00-98e4-11e9-8306-a5e9f66478f3.PNG)

[output_log.txt](https://github.com/shadowmage45/TexturesUnlimited/files/3336074/output_log.txt)

**Comment by shadowmage45** [comment]
Thanks for posting this.  I noticed during testing, but thought it was a stock issue.

I'll _attempt_ to investigate for a solution, but as with all things 'stock-icon-shader' related, the chances of me finding a solution without access to the stock shader source code is close to zero.

**Comment by Joker58th** [comment]
Your welcome.  Also, just an FYI, when I noticed this in my career, I also noticed the same problem with KIS containers.  If that helps at all.

**Comment by shadowmage45** [comment]
Interesting -- I wonder if somehow the stock code is swapping shaders out when rendering those icons, or what kind of stuff it is doing to the screen-coords that are passed into the shader.

Really wish I could see the source of the stock icon-shaders, as this would then be a very simple fix.

**Comment by shadowmage45** [comment]
Yeah, SQUAD is side-loading to create the icons for these parts, in a place where I cannot manipulate them.

More specifically, instead of using the existing part-icon from the prefab, which TU has already patched with the correct icon-shaders, they create a **NEW** icon anytime a new part is added to an inventory.  This new icon uses the default stock icon shaders, which are broken on DX11/OpenGL.

TLDR:  Don't think I can fix unless stock provides hooks for updating shaders whenever they create new icons.  This would be solvable with reflection and sufficient hackery, but I am not allowed to use reflection as-per the EULA.

**Comment by Joker58th** [comment]
Understood, thanks for looking.  Do I close this or do you or do we leave it open in case others look?  Sorry, I don't yet understand etiquette on these git issues as yet.

**Comment by shadowmage45** [comment]
I personally like to leave issues like this open, as it makes it more visible as a 'known issue', and there is always a potential that I'll think up some crazy solution or stock actually includes the needed hooks in future versions.

Granted, I may feel differently if I had lots of 'can't fix' issues, but generally there is only one or two per mod that pop up, so I don't mind leaving them on the open issues list.

**Comment by JamesB84** [comment]
Is there a temporary work-around for this issue? Is there something else besides DX11 or OpenGL I should use? FYI, DX12 doesn't seem to work either. It's not game breaking, but it would be nice to see the full icon in the inventory slots. Thanks!

**Comment by shadowmage45** [comment]
Nope, nothing can be done on this unless/until SQUAD releases the source for their icon shaders (or fixes their icon shaders for DX11).

The latter should be done for KSP 1.8 for DX11, but I'm personally not very optimistic that it will work properly for any custom shaders -- so the problem might be improved, but I doubt it will be fixed.

**Comment by shadowmage45** [comment]
Should be fixed with the latest releases.
